api/src/pcapi/core/educational/validation.py

```diff
@@ -98,7 +98,8 @@ def check_confirmation_limit_date_has_not_passed(booking: models.CollectiveBooki
 
 def check_collective_stock_is_editable(stock: models.CollectiveStock) -> None:
     offers_validation.check_validation_status(stock.collectiveOffer)
-    offers_validation.check_event_expiration(stock)
+    if stock.collectiveBookings:
```

Comment:
Attentinon un collectiveStock peut avoir un/des collectiveBooking/s. Il faut vérifier si il y en a au moins un qui ne soit pas au status Cancelled. Désolé j'ai oublié de le préciser dans la strat tech.

Reply:
De ce que je comprend, si collectivebookings cancelled -> PAS D'ERREUR, ce qui donnerai un truc du genre : 
```
if booking.status is not models.CollectiveBookingStatus.CANCELLED:
    offers_validation.check_event_expiration(stock)
```

Le truc c'est que les autres status pose erreur SAUF PENDING a ma connaissance, du coup on pourrai aussi mettre:
`if booking.status is models.CollectiveBookingStatus.PENDING:`
MAIS toujours d'apres ma comprehension on pourrai juste refacto ce code car la conditionnel du code pending serai avant .

Quel est la relation entre les dates et le status. y a il une relation d'un point de vue metier?cela me parait aberant de faire par code le fonction check_event_expiration() qui leve une erreur au niveau des dates si les dates ne sont pas coordonné et je n'arrive pas a comprendre ce que l'on recherche a faire a ce niveau.
pourquoi vérifier les dates si c'est cancel quoiqu'il en soit ( en admettant qu'on doivent le faire)? pourquoi les vérifier si c'est pending? ( en admettant qu'on doivent le faire )